src/Features/Core/Portable/ReplaceConditionalWithStatements/AbstractReplaceConditionalWithStatementsCodeRefactoringProvider.cs

```diff
@@ -0,0 +1,334 @@
+        //
+        //      ... a ? b : c ...;
+        //
+        // into
+        //
+        //      if (a)
+        //          ... b ...;
+        //      else
+        //          ... c ...;
+        //
+        // (in other words, we duplicate the statement wholesale, just replacing the conditional with the
+        // when-true/false portions.
+        var parentStatement = topExpression.Parent as TStatementSyntax;
+        if (IsSupportedSimpleStatement(syntaxFacts, parentStatement))
+        {
+            context.RegisterRefactoring(CodeAction.Create(
+                FeaturesResources.Replace_conditional_expression_with_statements,
+                c => ReplaceConditionalExpressionInSingleStatementAsync(document, conditionalExpression, parentStatement, c)),
+                conditionalExpression.Span);
+            return;
+        }
+
+        // However, if we're parented by a local decl, e.g.:
+        //
+        //      object v = a ? b : c;
+        //
+        // Then we want to break this into two statements.  One for the declaration, and one for the if-statement:
+        //
+        //      object v;
+        //      if (a)
+        //          v = b;
+        //      else
+        //          v = c;
+
+        if (topExpression.Parent is TEqualsValueClauseSyntax equalsValue &&
+            equalsValue.Parent is TVariableDeclaratorSyntax variableDeclarator &&
+            conditionalExpression.GetAncestor<TLocalDeclarationStatementSyntax>() is { } localDeclarationStatement &&
+            HasSingleVariable(localDeclarationStatement, out var variable) &&
+            syntaxFacts.IsDeclaratorOfLocalDeclarationStatement(variableDeclarator, localDeclarationStatement) &&
+            CanRewriteLocalDeclarationStatement(localDeclarationStatement))
+        {
+            context.RegisterRefactoring(CodeAction.Create(
+                FeaturesResources.Replace_conditional_expression_with_statements,
+                c => ReplaceConditionalExpressionInLocalDeclarationStatementAsync(
+                    document, conditionalExpression, variable, localDeclarationStatement, c)),
+                conditionalExpression.Span);
+            return;
+        }
+    }
+
+    private static TExpressionSyntax GetTopExpression(TConditionalExpressionSyntax conditionalExpression)
+    {
+        TExpressionSyntax current = conditionalExpression;
+
+        while (true)
+        {
+            if (current.Parent is TExpressionSyntax parentExpression)
+            {
+                current = parentExpression;
+                continue;
+            }
+
+            if (current.Parent is TArgumentSyntax { Parent: TArgumentListSyntax { Parent: TExpressionSyntax argumentParent } })
+            {
+                current = argumentParent;
+                continue;
+            }
+
+            return current;
+        }
+    }
+
+    private static SyntaxNode TryConvert(
```

Comment:
I guess this is a nit but can you move this down to where it's called?

Reply:
Sure.  will do in followup PR :)